liblangutil/Exceptions.h

```diff
@@ -139,21 +147,64 @@ class Error: virtual public util::Exception
 	static Error const* containsErrorOfType(ErrorList const& _list, Error::Type _type)
 	{
 		for (auto e: _list)
-		{
 			if (e->type() == _type)
 				return e.get();
-		}
 		return nullptr;
 	}
-	static bool containsOnlyWarnings(ErrorList const& _list)
+
+	static Severity errorSeverity(Type _type)
+	{
+		if (_type == Type::Info)
+			return Severity::Info;
+		if (_type == Type::Warning)
+			return Severity::Warning;
+		return Severity::Error;
+	}
+
+	static bool isError(Severity _severity)
+	{
+		return _severity == Severity::Error;
+	}
+
+	static bool isError(Type _type)
+	{
+		return isError(errorSeverity(_type));
+	}
+
+	static bool containsErrors(ErrorList const& _list)
 	{
 		for (auto e: _list)
+			if (isError(e->type()))
+				return true;
+		return false;
+	}
+
+	static std::string formatErrorSeverity(Severity _severity)
+	{
+		if (_severity == Severity::Info)
+			return "Info";
+		if (_severity == Severity::Warning)
+			return "Warning";
+		solAssert(isError(_severity), "");
+		return "Error";
+	}
+
+	static std::string formatErrorSeverityLowercase(Severity _severity)
+	{
+		switch (_severity)
 		{
-			if (e->type() != Type::Warning)
-				return false;
+		case Severity::Info:
+			return "info";
+		case Severity::Warning:
+			return "warning";
+		case Severity::Error:
+			[[fallthrough]];
+		default:
+			solAssert(isError(_severity), "");
+			return "error";
```

Comment:
There are only three possibilities so I'd drop the `default` case.

With a `default`, the compiler will not point out an error here when we add a new entry to the enum.